versioning/BRM/extentmap.cpp

```diff
@@ -3436,73 +3234,71 @@ void ExtentMap::createDictStoreExtent(int OID,
 //   segmentNum   - segment number to be assigned to the new extent
 // returns starting LBID of the created extent.
 //------------------------------------------------------------------------------
-LBID_t ExtentMap::_createDictStoreExtent(uint32_t size, int OID,
-        uint16_t  dbRoot,
-        uint32_t  partitionNum,
-        uint16_t  segmentNum)
+LBID_t ExtentMap::_createDictStoreExtent(uint32_t size, int OID, uint16_t dbRoot, uint32_t partitionNum,
+                                         uint16_t segmentNum)
 {
-    auto lastIndEmptyIndEmptyInd = _createExtentCommonSearch(OID, dbRoot,
-        partitionNum, segmentNum);
-    LastExtentIndexT lastExtentIndex = lastIndEmptyIndEmptyInd.first;
-    EmptyEMEntry emptyEMEntry = lastIndEmptyIndEmptyInd.second;
 
-    if (emptyEMEntry == -1)
+    uint32_t highestOffset = 0;
```

Comment:
IMHO we should revert this chunk and apply Index search here b/c we increase the complexity from O(1) to O(N) in the worst case.